library/psa_crypto.c

```diff
@@ -6424,13 +6419,14 @@ static psa_status_t psa_pbkdf2_set_input_cost(
         return PSA_ERROR_BAD_STATE;
     }
 
-    if (data > 0xFFFFFFFF) {
+    if (data > PSA_VENDOR_PBKDF2_MAX_ITERATIONS) {
```

Comment:
Please add a test case with good data except that the cost is `PSA_VENDOR_PBKDF2_MAX_ITERATIONS + 1`.

As things stand, we don't want a test case to ensure that `PSA_VENDOR_PBKDF2_MAX_ITERATIONS` does work, because that's unacceptably slow. I wonder if the default value of should be smaller. But probably not, because the largest value that can be sensible in a high-security, high-performance server setting would be insane on an embedded device.